core/src/spotify_id.rs

```diff
@@ -53,11 +68,22 @@ impl SpotifyId {
         let high = BigEndian::read_u64(&data[0..8]);
         let low = BigEndian::read_u64(&data[8..16]);
 
-        Ok(SpotifyId(u128::from_parts(high, low)))
+        Ok(SpotifyId::as_track(u128::from_parts(high, low)))
+    }
+
+    pub fn from_uri(uri: &str) -> Result<SpotifyId, SpotifyIdError> {
+        let parts = uri.split(":").collect::<Vec<&str>>();
+        if uri.contains(":show:") || uri.contains(":episode:") {
```

Comment:
Looks like `spotify:collection:podcasts:episodes` is a valid form too :)

https://github.com/librespot-org/librespot/pull/339#issuecomment-505139197

https://github.com/librespot-org/librespot/pull/339/commits/25bfc0582da61e29675474b7c516a1c6de45b289

Reply:
Can we also only have one implementation of this check? Seems to be in two places.

Reply:
Pushed some refactoring that should explain why I wasn't seeing the issue with `collection:podcasts`